bin/stacks/routing-lambda-stack.ts

```diff
@@ -77,16 +87,79 @@ export class RoutingLambdaStack extends cdk.NestedStack {
     new CfnOutput(this, 'jsonRpcProviders', {
       value: JSON.stringify(jsonRpcProviders),
     })
+    const stage = process.env.STAGE || 'dev' // Default to 'dev' if not set
+
+    // TODO: Add if not dev , not create
+    const vpc = new ec2.Vpc(this, `RoutingLambdaVPC-${stage}`, {
+      maxAzs: 2, // Number of availability zones
+      subnetConfiguration: [
+        {
+          name: `RoutingPublicSubnet-${stage}`,
+          subnetType: ec2.SubnetType.PUBLIC, // Public subnet with internet access
+          cidrMask: 24, // IP range for public subnet
+        },
+        {
+          name: `RoutingPrivateSubnet-${stage}`,
+          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS, // Private subnet with access to the internet via NAT Gateway
+          cidrMask: 24, // IP range for private subnet
+        },
+      ],
+      natGateways: 1, // One NAT Gateway for private subnet internet access
```

Comment:
not sure if related but good to mention here:
on the unirpc side we started hitting a port connections limit once i started ramping up traffic for routing api
It's weird that routing api lambda didn't hit similar issue, as it is doing similar size of http calls to QN directly.
After further digging, I see that routing api default VPC doesn't have a NAT gateway (and can't see active connections dashboards)?
But with this change we are introducing new VPC with NAT gateway, so trying to understand if that might cause issues due to high amount of traffic.
Anyway we could test the new vpc introduction somehow before fully enabling?